epp-client/src/connection/client.rs

```diff
@@ -177,4 +177,10 @@ impl EppClient {
     pub fn greeting(&self) -> Result<EppGreeting, error::Error> {
         EppGreeting::deserialize(&self.connection.greeting)
     }
+
+    /// A utility function to generate a simple client TRID
+    pub fn generate_client_tr_id(&self) -> Result<String, Box<dyn Error>> {
```

Comment:
You removed the comment about this being used mainly for testing and that users should invent their own. Unless the specification requires that transaction IDs include the username, I think keeping this as a free function makes more sense, and it should probably not be part of the public API (integration tests could just use `"test-1"` or some such).

Reply:
I removed any internal reference to this function but left it available since it seems useful and relevant to me. I could re-add the disclaimer but I don't think it's unreasonable to use this from the caller.

Reply:
I ended up removing this completely. See comment above

Reply:
This can be removed. It wasn't meant to be used by callers, and initially I was checking if callers could pass in a function when creating an `EppClient` variable to generate this `trID` since in most cases they would need a function that generates a unique value each time. But passing in a function is a little awkward so would be best to just expect them to pass in the trID string after generating it in whatever way they want.